common/src/test/java/com/revenuecat/purchases/common/verification/SigningManagerTest.kt

```diff
@@ -0,0 +1,99 @@
+package com.revenuecat.purchases.common.verification
+
+import android.util.Base64
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.common.networking.HTTPResult
+import io.mockk.every
+import io.mockk.mockk
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class SigningManagerTest {
+    private companion object {
+        const val TEST_SIGNATURE = "Jmax3TdnBIe0/zFeHT5KJrFNoGxWtQAOuYTjnEXDHa0z3/npDG9nRB4vrUkt/ZxVh7SU+P++O3LnObxeuz3tFAILs75bxIqXwp6LqdV7Tgo="
+        const val TEST_NONCE = "NYwvC+KDXzUq9pfg"
+    }
+
+    private lateinit var verifier: SignatureVerifier
+
+    private lateinit var signingManager: SigningManager
+
+    @Before
+    fun setUp() {
+        verifier = mockk()
+
+        signingManager = SigningManager(verifier)
+    }
+
+    // region createRandomNonce
+
+    @Test
+    fun `createRandomNonce returns 12 base64 encoded random bytes`() {
```

Comment:
I added another test in iOS to verify that 2 created nonces are different (as a way to verify randomness).